CMakeLists.txt

```diff
@@ -619,6 +619,14 @@ if(USE_LIBIDN2)
   check_library_exists_concat("idn2" idn2_lookup_ul HAVE_LIBIDN2)
 endif()
 
+if(WIN32)
+  option(USE_WIN32_IDN "Use WinIDN for IDN support" OFF)
+  if(USE_WIN32_IDN)
+    list(APPEND CURL_LIBS "Normaliz")
+    add_definitions(-DUSE_WIN32_IDN -DWANT_IDN_PROTOTYPES)
```

Comment:
Shouldn't this rather use `set()` to define those defines and have them in `lib/curl_config.h.cmake` ? That would at least make it much more similar to configure does it and how we set a lot of other defines for the build.

Reply:
Thank you!
According to your comment, I have pushed a new commit, please review it.